addons/mrp/models/mrp_abstract_workorder.py

```diff
@@ -212,6 +217,16 @@ def _update_raw_moves(self):
         self.workorder_line_ids.unlink()
         self.env['stock.move.line'].create(vals_list)
 
+    def _check_strict_consumption(self, lines_to_check):
+        for line in lines_to_check:
+            same_pp_line = self.workorder_line_ids.filtered(lambda l: l.product_id == line.product_id)
+            qty_done = sum(same_pp_line.mapped('qty_done'))
+            qty_to_consume = sum(same_pp_line.mapped('qty_to_consume'))
+            rounding = self.product_uom_id.rounding
+            if float_compare(qty_done, qty_to_consume, precision_rounding=rounding) == 1:
+                raise UserError(_('You should consume the quantity of components defined in the BoM. If you want to consume more or less components, change the consumption setting on the BoM.'))
+        return lines_to_check
```

Comment:
if you group self.workorder_line_ids by move you'll handle the bom with multiple times the same products and diffeent quantities